frontends/concrete-python/concrete/fhe/mlir/processors/assign_bit_widths.py

```diff
@@ -88,13 +88,38 @@ def max_encrypted_bitwidth_node(node: Node):
         return normal_p + 1
 
     if name in MULTIPLY and all(value.is_encrypted for value in node.inputs):
-        return normal_p + 1
+        # For operations that use multiply, an additional bit
+        # needs to be added to the bitwidths of the inputs.
+        # For single precision circuits the max of the input / output
+        # precisions will be taken in required_encrypted_bitwidth. For
+        # multi-precision, the circuit partitions will handle the
+        # input and output precisions separately.
+        all_inp_bitwidths = []
+        # Need a loop here to allow typechecking and make mypy happy
```

Comment:
mypy is not deciding how you code. you can locally disable it if needed.
You can `# type: ignore` a particular line.
You can also use https://docs.python.org/3/library/typing.html#typing.cast

Reply:
Sure, but in this case a loop also makes the code more clear. I prefer to disable it only when there is no choice or the alternative code is worse.